### PR TITLE
Feature/health check v ms

### PR DESCRIPTION
**Pull request by jweite-amazon**
*Issue #, if available:*

*Description of changes:*
Added templates for adding control plane and worker machine health checks.  These are intentionally segregated from the provided clusterTemplates, for use by deployers as they see fit.

*Testing performed:*
Manual testing of unhealthy detection and remediation by stopping ACS VMs and monitoring the remediation process.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rejoshed — APPROVED** [review]
Could see putting this in one file, but I'm happy either way. Looks good!

**Comment by rejoshed** [comment]
Can ltgm if you'd like, or feel free to add the label yourself. : )

File: templates/controlPlaneMachineHealthCheck.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+metadata:
+  name: ${CLUSTER_NAME}-kcp-unhealthy-2m
+spec:
+  clusterName: ${CLUSTER_NAME}
+  maxUnhealthy: 100%
+#  nodeStartupTimeout: 10m
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/control-plane: ""
+  unhealthyConditions:
```
Comment (rejoshed) [inline templates/controlPlaneMachineHealthCheck.yaml:25]: Kidna weird that we're using the negative here. Would positive health checks be more complete? Is that possible?

Comment (jweite-amazon) [reply templates/controlPlaneMachineHealthCheck.yaml:25]: That's just how the CAPI MachineHealthCheck CRD is defined: https://github.com/kubernetes-sigs/cluster-api/blob/58905fb5140a7844c3e24aa70d255fb8f9d0c491/api/v1alpha3/machinehealthcheck_types.go#L41

**Comment by maxdrib** [comment]
Why not include these objects by default, and segregate them instead?

**Comment by jweite-amazon** [comment]
@maxdrib, I don't follow how to include-yet-segregate.  What did you have in mind?

**Comment by maxdrib** [comment]
> @maxdrib, I don't follow how to include-yet-segregate. What did you have in mind?

I just mean why are we segregating them? Why not put them in the default templates? Are there any downsides of including this as a default object installed on a CAPC cluster?

**Comment by jweite-amazon** [comment]
@maxdrib, comments in the file elaborate on why I elected to segregate.  Most significantly is that "certain CNIs" (I didn't want to call out Cilium by name) interfere with the deletion of a failed worker node.  I don't want to cause that to happen for folks who may not fully understand what's the mechanism.  Instead I prefer to let them explicitly elect to try health checking and then discover whether it's compatible with their CNI or not.

**Pull request merged**